specs/fault-proof/cannon-fault-proof-vm.md

```diff
@@ -218,3 +221,49 @@ transition. Nominally, the FPVM must raise an exception in at least the followin
 VM implementations may raise an exception in other cases that is specific to the implementation.
 For example, an on-chain FPVM that relies on pre-supplied merkle proofs for memory access may
 raise an exception if the supplied merkle proof does not match the pre-state `memRoot`.
+
+## Security Model
+
+### Compiler Correctness
+
+Cannon is designed to prove the correctness of a particular state transition that emulates a MIPS32 machine.
+Cannon does not guarantee that the MIPS32 instructions correctly implement the program that the user intends to prove.
+As a result, Cannon's use as a Fault Proof system inherently depends on some extent on the correctness of the compiler
+used to generate the MIPS32 instructions over which Cannon operates.
+
+To illustrate this concept, suppose that a user intends to prove simple program `input + 1 = output`.
+Suppose then that the user's compiler for this program contains a bug and errantly generates the MIPS instructions for a
+slightly different program `input + 2 = output`. Although Cannon would correctly prove the operation of this compiled program,
+the result proven would differ from the user's intent. Cannon proves the MIPS state transition but makes no assertion about
+the correctness of the translation between the user's high-level code and the resulting MIPS program.
```

Comment:
It feels a bit weird to me to call out bugs in compilers specifically for cannon when the exact same risk exists for native op-node and op-geth but isn't called out in the spec.

Reply:
Realistically I would prefer that we call out every risk as explicitly as possible, whenever possible. I do think the risk is somewhat unique here as people may not already have built up a mental model that makes the compiler risk obvious. I wouldn't be surprised if the compiler risk is non-obvious to a lot of people.

Reply:
For this specific case, I think it's worth while. We don't call out compiler bugs in our standard native software, yes, but the MIPS target is the least-actively-maintained target in the Go compiler I'd bet. The risk is substantially higher than average native Go code.